src/CircuitZKit.ts

```diff
@@ -0,0 +1,254 @@
+
+      this._moveFromTempDirToOutDir(tempDir, artifactDir);
+    } finally {
+      fs.rmSync(tempDir, { recursive: true, force: true });
+    }
+  }
+
+  public async createVerifier(): Promise<void> {
+    const tempDir = this._manager.getTempDir();
+
+    try {
+      const verifierDir = this._getDir("verifier");
+
+      fs.mkdirSync(tempDir, { recursive: true });
+
+      const vKeyFile = this._mustGetFile("vkey");
+      const verifierFile = this._getFile("sol", tempDir);
+
+      const groth16Template = this._manager.getTemplate("groth16");
+
+      const templateParams = JSON.parse(fs.readFileSync(vKeyFile, "utf-8"));
+      templateParams["verifier_id"] = this.getVerifierId();
+
+      const verifierCode = ejs.render(groth16Template, templateParams);
+
+      fs.writeFileSync(verifierFile, verifierCode, "utf-8");
+
+      this._moveFromTempDirToOutDir(tempDir, verifierDir);
+    } finally {
+      fs.rmSync(tempDir, { recursive: true, force: true });
+    }
+  }
+
+  public async generateProof(inputs: Inputs): Promise<ProofStruct> {
+    const zKeyFile = this._mustGetFile("zkey");
+    const wasmFile = this._mustGetFile("wasm");
+
+    return (await snarkjs.groth16.fullProve(inputs, wasmFile, zKeyFile)) as ProofStruct;
+  }
+
+  public async verifyProof(proof: ProofStruct): Promise<boolean> {
+    const vKeyFile = this._mustGetFile("vkey");
+
+    const verifier = JSON.parse(fs.readFileSync(vKeyFile).toString());
+
+    return await snarkjs.groth16.verify(verifier, proof.publicSignals, proof.proof);
+  }
+
+  public async generateCalldata(proof: ProofStruct): Promise<Calldata> {
+    const calldata = await snarkjs.groth16.exportSolidityCallData(proof.proof, proof.publicSignals);
+
+    return JSON.parse(`[${calldata}]`) as Calldata;
+  }
+
+  public getCircuitId(): string {
+    return path.parse(this._circuit).name;
+  }
+
+  public getVerifierId(): string {
+    return `${path.parse(this._circuit).name}Verifier`;
+  }
+
+  private async _generateZKey(outDir: string): Promise<void> {
+    const r1csFile = this._getFile("r1cs", outDir);
+    const zKeyFile = this._getFile("zkey", outDir);
+
+    const constraints = await this._getConstraints(outDir);
+    const ptauInfo = await this._manager.fetchPtauInfo(constraints);
+
+    await snarkjs.zKey.newZKey(r1csFile, ptauInfo.file, zKeyFile);
+  }
+
+  private async _generateVKey(outDir: string): Promise<void> {
+    const zKeyFile = this._getFile("zkey", outDir);
+    const vKeyFile = this._getFile("vkey", outDir);
+
+    const vKeyData = await snarkjs.zKey.exportVerificationKey(zKeyFile);
+
+    fs.writeFileSync(vKeyFile, JSON.stringify(vKeyData));
+  }
+
+  private _getCompileArgs(options: CompileOptions, outDir: string): string[] {
+    let args = [this._circuit, "--r1cs", "--wasm"];
+
+    options.sym && args.push("--sym");
+    options.json && args.push("--json");
+    options.c && args.push("--c");
+    !options.quiet && args.push("--verbose");
```

Comment:
So if it's not quiet it's verbose by default?

Reply:
Rm verbose for now, I see no difference in output tbh